src/handlers/track_table.rs

```diff
@@ -149,6 +152,108 @@ pub fn handler(key: Key, app: &mut App) {
   }
 }
 
+fn play_random_song(app: &mut App) {
+  let TrackTable { context, .. } = &app.track_table;
+  match &context {
+    Some(context) => match context {
+      TrackTableContext::MyPlaylists => {
+        let (context_uri, track_json) = match (&app.selected_playlist_index, &app.playlists) {
+          (Some(selected_playlist_index), Some(playlists)) => {
+            if let Some(selected_playlist) = playlists.items.get(selected_playlist_index.to_owned())
+            {
+              (
+                Some(selected_playlist.uri.to_owned()),
+                selected_playlist.tracks.get("total"),
+              )
+            } else {
+              (None, None)
+            }
+          }
+          _ => (None, None),
+        };
+
+        if let Some(val) = track_json {
+          let num_tracks: usize = from_value(val.clone()).unwrap();
+          app.dispatch(IoEvent::StartPlayback(
+            context_uri,
+            None,
+            Some(thread_rng().gen_range(0, num_tracks)),
+          ));
+        }
+      }
+      TrackTableContext::RecommendedTracks => {}
+      TrackTableContext::SavedTracks => {
+        if let Some(saved_tracks) = &app.library.saved_tracks.get_results(None) {
+          let track_uris: Vec<String> = saved_tracks
+            .items
+            .iter()
+            .map(|item| item.track.uri.to_owned())
+            .collect();
+          let rand_idx = thread_rng().gen_range(0, track_uris.len());
+          app.dispatch(IoEvent::StartPlayback(
+            None,
+            Some(track_uris),
+            Some(rand_idx),
+          ))
+        }
+      }
+      TrackTableContext::AlbumSearch => {}
+      TrackTableContext::PlaylistSearch => {
+        let (context_uri, playlist_track_json) = match (
```

Comment:
This can be simplified too 

```rust 
        if let (Some(selected_search_playlist_index), Some(search_playlists)) = (
          app.search_results.selected_playlists_index,
          &app.search_results.playlists,
        ) {
          if let Some(selected_playlist) = search_playlists
            .playlists
            .items
            .get(selected_search_playlist_index.to_owned())
          {
            let num_tracks = search_playlists.playlists.total as usize;
            let context_uri = selected_playlist.uri.clone();
            app.dispatch(IoEvent::StartPlayback(
              Some(context_uri),
              None,
              Some(thread_rng().gen_range(0, num_tracks)),
            ))
          };
        };

```

Reply:
`let num_tracks = search_playlists.playlists.total as usize;` get the total number of playlists, I'm looking for the total number of tracks under a playlist.